ax/service/ax_client.py

```diff
@@ -1255,29 +1257,11 @@ def fit_model(self) -> None:
         If no model is yet available a new one is instantiated. This
         may be the case when get_next_trial() has never been called.
         """
-
-        # Try to instantiate a model if there is none. This
-        # handles the case where trials have not be generated by calling
-        # get_next_trial().
-        if self.generation_strategy.model is None:
-            logger.info(
-                "get_model_predictions() has been called when no model is "
-                "instantiated. Attempting to instantiate the model for the "
-                "first time."
+        if not self.experiment.trial_indices_by_status[TrialStatus.COMPLETED]:
+            raise DataRequiredError(
+                "At least one trial must be completed with data to fit a model."
             )
-            if not self.experiment.trial_indices_by_status[TrialStatus.COMPLETED]:
-                raise ValueError(
-                    "At least one trial must be completed with data to instantiate "
-                    "a model."
-                )
-            self.generation_strategy._fit_current_model(data=None)
-            logger.info("Successfully instantiated a model for the first time.")
-
-        # Model update is normally tied to the GenerationStrategy.gen() call,
-        # which is called from get_next_trial(). In order to ensure that predictions
-        # can be performed without the need to call get_next_trial(), we update the
-        # model with all attached data. Note that this method keeps track of previously
-        # seen trials and will update the model if there is newly attached data.
+        self.generation_strategy._maybe_transition_to_next_node()
```

Comment:
Why might we need to transition to the next node?

Reply:
This could go either way. With this call, we fit the model that would be used to generate the next trial. Without it, we would just re-train (with updated data) the model that was used to fit the last trial. I'll add a note about this to the docstring.